### PR TITLE
Tabs support badge

### PR DESCRIPTION
**Pull request by plowsirius**
#1536  https://github.com/ant-design/ant-design-mobile/pull/1568 https://github.com/ant-design/ant-design-mobile/pull/1601

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1604)

### CODE REVIEW

**Comment by paranoidjk** [comment]
ref https://github.com/ant-design/ant-design-mobile/pull/1601

**Comment by paranoidjk** [comment]
@plowsirius 样式需要再调整下，多测试。

File: components/tabs/style/index.less [hunk 1]
```diff
@@ -69,10 +69,28 @@
...
       box-sizing: border-box;
+      align-items: center;
 
       &-active {
         color: @tabs-color;
       }
+
+      .am-badge {
+        .am-badge-text {
+          top: -0.26rem;
```
Comment (paranoidjk) [inline components/tabs/style/index.less]: less 源码用 px 为单位，构建的时候 postcss  会处理转换

File: components/tabs/style/index.less [hunk 2]
```diff
@@ -69,10 +69,28 @@
       color: @color-text-base;
       background-color: @fill-base;
       box-sizing: border-box;
+      align-items: center;
 
       &-active {
         color: @tabs-color;
       }
+
+      .am-badge {
```
Comment (paranoidjk) [inline components/tabs/style/index.less]: 定义一个变量 badge_prefix_class, 然后 less 里面用 &

还是一个目的，代码的可维护性，万一以后 badge 的 prefixCls 变了 😄

File: components/tabs/style/index.less [hunk 3]
```diff
@@ -69,10 +69,28 @@
...
+          left: 100%;
+          right: auto;
+          transform: translateX(-0.1rem);
+          transform-origin: center;
+        }
+        .am-badge-dot {
+          top: -0.12rem;
+          left: 100%;
+          right: auto;
+          transform: translateX(0);
```
Comment (paranoidjk) [inline components/tabs/style/index.less]: 这行的目的是什么？

**Comment by pingan1927** [comment]
截图上边界超出的问题，这个业务上自己考虑比较合适，确实需要在边上的Tab加很长的Badge，业务上自行覆盖样式。

File: components/tabs/style/index.less (hunk 3 above)
Comment (pingan1927) [inline components/tabs/style/index.less]: 样式更好看点。

**Pull request merged**